sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/reflection/KalixMethod.scala

```diff
@@ -0,0 +1,197 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.springsdk.impl.reflection
+
+import com.google.protobuf.Descriptors
+import kalix.springsdk.impl.path.{ PathPattern, PathPatternParser }
+import kalix.springsdk.impl.reflection.RestServiceIntrospector.{
+  isEmpty,
+  validateRequestMapping,
+  PathParameter,
+  RestMethodParameter
+}
+import org.springframework.web.bind.annotation.{ RequestMapping, RequestMethod }
+
+import java.lang.reflect.Method
+import scala.annotation.tailrec
+
+sealed trait ServiceMethod {
+  def methodName: String
+  def javaMethodOpt: Option[Method]
+
+  def requestMethod: RequestMethod
+  def pathTemplate: String
+}
+
+sealed trait AnyServiceMethod extends ServiceMethod {
+  def inputType: Class[_]
+
+  protected def buildPathTemplate(componentName: String, methodName: String): String = {
+    val cls = componentName.replace("$", ".")
+    s"/$cls/${methodName.capitalize}" // FIXME: must pass through NameGenerator?
+  }
+}
+
+/**
+ * Build from methods annotated with @Subscription at type level.
+ *
+ * It's used as a 'virtual' method because there is not Java method backing it. It will exist only in the gRPC
+ * descriptor and will be used for view updates with transform = false
+ */
+case class VirtualServiceMethod(component: Class[_], methodName: String, inputType: Class[_]) extends AnyServiceMethod {
+
+  override def requestMethod: RequestMethod = RequestMethod.POST
+
+  override def javaMethodOpt: Option[Method] = None
+
+  val pathTemplate = buildPathTemplate(component.getName, methodName)
+}
+
+/**
+ * Build from methods annotated with @Subscription. Those methods are not annotated with Spring REST annotations, but
+ * they become a REST method at the end.
+ */
+case class RestServiceMethod(javaMethod: Method) extends AnyServiceMethod {
+
+  val inputType: Class[_] = javaMethod.getParameterTypes()(0)
+
+  override def methodName: String = javaMethod.getName
+
+  override def requestMethod: RequestMethod = RequestMethod.POST
+  override def javaMethodOpt: Option[Method] = Some(javaMethod)
+
+  val pathTemplate = buildPathTemplate(javaMethod.getDeclaringClass.getName, methodName)
+}
+
+/**
+ * Build from Spring annotations
+ *
+ * @param callable
+ *   Is this actually a method that will ever be called or just there for metadata annotations?
+ */
+case class SpringRestServiceMethod(
+    classMapping: Option[RequestMapping],
+    mapping: RequestMapping,
+    javaMethod: Method,
+    params: Seq[RestMethodParameter],
+    callable: Boolean = true)
```

Comment:
In which case will that be false? 

As far I understand it, methods that only exist for metadata are covered by `VirtualServiceMethod`. Those are synthetic methods that we will create if the `@Subscription` annotation is added at type level, meaning that there is no transformation and therefore no method to call in the user function.